Legifrance.js

```diff
@@ -40,252 +40,261 @@ Thanks to Sebastian Karcher and Aurimas Vinckevicius
-		}
-
-		var g; // tribunaux administratifs sans chambre avec publication
-		g = title.match(/(Tribunal Administratif|administratif.*), du (.*), (s*[0-9-]+), (.*Lebon)/)
-		if (g) {
-			var cour = g[1];
-			var date = g[2];
-			var numero = g[3];
-			var publication = g[4];
-			newItem.court = 'Tribunal ' + cour;
-			newItem.date = date;
-			newItem.docketNumber = numero;
-			newItem.reporter = publication;
-		}
-
-		// Note : présence d'autres cas pour les TA
-
-		var h; // Cour de cassation 
-		h = title.match(/(Cour de cassation), (.*), (.*), (s*[0-9-. ]+), (.*)/)
-		if (h) {
-			var nature = h[1];
-			var formation = h[2];
-			var date = h[3];
-			var numero = h[4];
-			var publication = h[5];
-			newItem.court = 'Cour de cassation';
-			if (nature) newItem.tags.push(nature);
-			newItem.extra = formation;
-			newItem.date = date;
-			newItem.docketNumber = numero;
-			newItem.reporter = publication;
-		}
-
-		var i; // cours d'appel et tribunaux
-		i = title.match(/(Cour d'appel.*|Tribunal.*|Conseil.*|Chambre.*|Juridiction.*|Commission.*|Cour d'assises.*) de (.*), (.*), (s*[0-9/]+)/)
-		if (i) {
-			var cour = i[1];
-			var lieu = i[2];
-			var date = i[3];
-			var numero = i[4];
-			newItem.court = cour + ' de ' + lieu;
-			newItem.date = date;
-			newItem.docketNumber = numero;
-		}
-
-		// Tribunal des conflits - Base CASS
-		j = title.match(/(Tribunal des conflits), (.*), (.*), (s*[0-9-. ]+), (.*)/)
-		if (j) {
-			var nature = j[2];
-			var date = j[3];
-			var numero = j[4];
-			var publication = j[5];
-			newItem.court = 'Tribunal des conflits';
-			if (nature) newItem.tags.push(nature);
-			newItem.date = date;
-			newItem.docketNumber = numero;
-			newItem.reporter = publication;
+function detectWeb(doc, url) {
+	if (url.match(/.CETATEXT|CONSTEXT|JURITEXT./)) { // Détection jurisprudence
+		return "case";
+	} else if (url.match(/LEGIARTI|affichCodeArticle|affichTexteArticle|KALICONT|JORFTEXT|CNILTEXT/)) { // Détection textes législatifs
+		return "statute"; // Détection lois et codes
+	} else if (url.match(/rechJuriConst|rechExpJuriConst|rechJuriAdmin|rechExpJuriAdmin|rechJuriJudi|rechExpJuriJudi/)) { // Détection occurences multiples uniquement pour la jurisprudence
+		return "multiple"; // occurences multiples
+	} else return false;
+}
 
-		}
-		newItem.complete();
+function scrapecase(doc) { //Jurisprudence
+	var newItem = new Zotero.Item("case");
+	
+	// Paramètres communs
+	var numero, date, formation, cour, publication, nature;
+	
+	var title = ZU.xpathText(doc, '//h2[@class="title"]');
+	newItem.title = title;
+	newItem.url = doc.location.href;
+	var rtfurl = ZU.xpathText(doc, '//a[contains(text(), "Télécharger")]/@href');
+	if (rtfurl) {
+		newItem.attachments = [{
+			url: "http://www.legifrance.gouv.fr/" + rtfurl,
+			title: "Document en RTF",
+			mimeType: "application/rtf"
+		}];
+	}
+	
+	// Situation selon les juridictions
+	
+	var a;// Conseil constitutionnel
+	a = title.match(/(.*) - (.*) - (.*) - (.*)/);
```

Comment:
no need to separately declare. Just do `const a = title.match()` and similarly below